src/Display/Display.cpp

```diff
@@ -1,3 +1,32 @@
 #include "Display.h"
 
-Display gDisplay;
+// ODROID
+#ifdef __XTENSA__
+#include "OdroidDisplay.h"
+OdroidDisplay child;
```

Comment:
"child" is going to be a global variable accessible from other object files.

The proper way to do this is static ODroidDIsplay child;

This makes it local only to the original file.

That said, my idea was more something like:

```
Class Display {
// this is an abstract class (read up on virtual methods with =0)
};
```
Then in LDKDisplay.h:
```
class LDKDisplay : public Display {
// implements Display methods
};
```
In DesktopDIsplay.h"
```
class DesktopDisplay : public Display {
// implements Display methods
};
```

The conditional ifdefs are correct, but within, you'd have
```
LDKDisplay gDisplay;

or

DesktopDisplay gDisplay;
```

The includes should be consistent, probably using "./LDKDisplay/LDKDisplay.h" and so on.

Reply:
Going to start a new branch and PR. That will be easier than salvaging this one.